sdk/samples/iot/templates/iothub/test-resources-post.ps1

```diff
@@ -17,6 +17,29 @@ param (
     $RemainingArguments
 )
 
+function waitForActiveHub
+{
+    $retryCount = 0
+    do
+    {
+      $retryCount++
+      Write-Host "AzIotHub is not yet active so sleeping for $retryCount seconds."
+      Start-Sleep -Seconds $retryCount
+
+      # Get the hub as an object
+      $hub_obj = Get-AzIotHub -ResourceGroupName $ResourceGroupName -Name $iothubName
+    }
+    while ($retryCount -lt 8 -and $hub_obj.Properties.State -ne "Active")
+
+    if ($hub_obj.Properties.State -ne "Active")
+    {
+      Write-Error "AzIotHub instance is not yet active so later steps might fail as a result. If they fail often you can increase the retry count above."
```

Comment:
`Write-Error` itself may result in the step failing ([default `errorActionPreference` is `stop`](https://docs.microsoft.com/en-us/azure/devops/pipelines/tasks/utility/powershell?view=azure-devops#arguments)). If you intend to warn here use a [DevOps formatting command](https://docs.microsoft.com/en-us/azure/devops/pipelines/scripts/logging-commands?view=azure-devops&tabs=bash#formatting-commands)

```suggestion
      Write-Host "##[warning]AzIotHub instance is not yet active so later steps might fail as a result. If they fail often you can increase the retry count above."
```

Reply:
Is there any chance operations below will pass if the hub isn't ready? Shouldn't we fail here?

Reply:
yes this is supposed to fail if the retries expire.